geomesa-feature/src/main/scala/org/locationtech/geomesa/feature/SimpleFeatureEncoder.scala

```diff
@@ -59,38 +59,85 @@ trait SimpleFeatureDecoder extends HasEncoding {
 
 object SimpleFeatureDecoder {
 
-  def apply(sft: SimpleFeatureType, encoding: FeatureEncoding) =
+  /**
+   * Decode without projecting.
+   *
+   * @param sft the encoded simple feature type to be decode
+   * @param encoding the encoding that was used to encode
+   * @param options any options that were used to encode
+   * @return a new [[SimpleFeatureDecoder]]
+   */
+  def apply(sft: SimpleFeatureType, encoding: FeatureEncoding, options: EncodingOption*) =
```

Comment:
The signitures are different:

apply(sft: SimpleFeatureType, encoding: FeatureEncoding, options: Set[EncodingOptions] = Set.empty)

apply(sft: SimpleFeatureType, encoding: String, options: Set[EncodingOptions] = Set.empty)

but the complier doesn't allow multiple overloaded methods with default arguments:

https://groups.google.com/forum/#!msg/scala-user/FyQK3-cqfaY/fXLHr8QsW_0J